mycpp/control_flow_pass.py

```diff
@@ -319,29 +333,14 @@ def visit_with_stmt(self, o: 'mypy.nodes.WithStmt') -> None:
 
         # Note: we have 'with alloc.ctx_SourceCode'
         #assert isinstance(expr.callee, NameExpr), expr.callee
-        callee_name = expr.callee.name
+        if isinstance(expr.callee, NameExpr):
+            callee_name = expr.callee.name
+            if callee_name in ('switch', 'str_switch', 'tagswitch'):
```

Comment:
ditto here, we shouldn't duplicate the logic

The way I think of it is that `visitor.py` is  creating a **"compiler IR"** for the "mycpp language" !

The IR (intermediate representation) now has `switch`, which Python does **not** have

And then `control_flow_pass` and `cppgen_pass` are **consuming** the IR.  But they shouldn't need to know the details -- that can be hidden


(We might want to move away from the visitor pattern and even away from the MyPy AST -- there are some discussions on Zulip about this)